### ISSUE TITLE
Run tests in parallel

### ISSUE THREAD

**Issue by robfletcher** [spockframework/spock#157]
Originally reported on Google Code with ID 34

```
I have an enhancement request to make tests run in parallel. I like Spock
so much, I'm using it to run some validation tests that run over a large
dataset. I use the data provider way and @Unroll to run the tests as unique
tests. They work great, except they run serially and I'd like to use up all
the cores that the machine has (> 4).
```

Reported by `senthilvasan` on 2009-07-17 21:42:40

**Comment by automatictester** [comment]
I can see this issue is still open and there is no mention of parallel execution in the official docs. When can the community expect this issue to be addressed? For those who use Spock for something else than unit testing this would be an important improvement. It would be a shame if lack of this feature would make people look for alternatives.

**Comment by gkastrinis** [comment]
This would be really helpful, especially (as in my case) when each iteration of a data-driven feature takes a couple of minutes to run.

**Comment by kinzlp** [comment]
Running though surefire is can do parallel now. 
That would be a great idea to have it support parallel testing, especially for data-driven test cases.
It would by even better if it can be controlled by annotation like @Unroll, @Parallel.

**Comment by szpak** [comment]
Related PR - #558.

**Comment by dengayevskiy** [comment]
Hi!

I'm using **Gradle + Spock** in my project for writing integration tests. At this stage, the test suite is running for about an hour and it's pretty slow ;( So to speed it up our my team decided to use tests parallelization. Gradle supports parallel tests execution, but only with help of forks and in this way I'm limited with the number of cores on my machine (or even less). 

Is there any terms or plans to add parallel tests execution to Spock?

**Comment by arjunankaruppusamy** [comment]
Hi!

I'm using Gradle + Spock in my project for writing integration tests. At this stage, the test suite is running for about an hour and it's pretty slow ;( So to speed it up our my team decided to use tests parallelization. Gradle supports parallel tests execution, but only with help of forks and in this way I'm limited with the number of cores on my machine (or even less).

Is there any terms or plans to add parallel tests execution to Spock?

**Comment by arjunankaruppusamy** [comment]
Hi!

I'm using Gradle + Spock in my project for writing integration tests. At this stage, the test suite is running for about an hour and it's pretty slow ;( So to speed it up our my team decided to use tests parallelization. Gradle supports parallel tests execution, but only with help of forks and in this way I'm limited with the number of cores on my machine (or even less).

Is there any terms or plans to add parallel tests execution to Spock?

gradle.properties:
org.gradle.workers.max=10
maxparellelforks=10

build.gradle:
test {
        maxParallelForks = project.maxparellelforks as int
}

It will work can you try this one. Make sure none other program should not run on you machine because while creating maxparallelforks will consume more cpu once created, will release cpu.

**Comment by dlehammer** [comment]
Hi @automatictester , @gkastrinis , @dengayevskiy , @arjunankaruppusamy & anyone else who thirsts for parallel execution in Spock.

Please consider directing your attention to [Support for parallel feature and iteration execution](https://github.com/spockframework/spock/pull/558) (Pull request #588).

As far as I can tell, this feature has been implemented - but requires testing and review - before it's released :-D
Some of you appear to have access to Spock test-suites that would be perfect for some real world testing of the [PR-branch](https://github.com/Fuud/spock/tree/parallel__over_spock_master).

Perhaps @Fuud or a maintainer could provide a Spock SNAPSHOT-binary or guidance/howto generate a local binary for testing.

**Comment by nandanbadheka** [comment]
DO we know when will this feature be released ? we can help testing too, if it will be helpful.

**Comment by jacattau** [comment]
Adding my +1 to get parallel execution

**Comment by leonard84** [comment]
Spock 2.0 will support parallel execution provided via the JUnit 5 Platform

**Comment by jacattau** [comment]
@leonard84 that's great and all but this issue has been open for 3.5 years.  Do we have a rough estimate as to when Spock 2.0 is coming? You also said something similar in another issue 6 months ago:

> Sorry but this will not be merged into Spock 1.x. Spock 2.0 will use the native support from the JUnit > 5 Plattform for parallel execution.

> _Originally posted by @leonard84 in > https://github.com/spockframework/spock/pull/558#issuecomment-417375642_

Spock in on 1.3 at the moment, so it seems 2.0 seems way out on the horizon.  I know that isn't how versions work, but we don't have any better indication.

**Comment by leonard84** [comment]
> Spock is on 1.3 at the moment, so it seems 2.0 seems way out on the horizon.

No, 1.3 is the last planned 1.x release. We have already started with the development for 2.0 https://github.com/spockframework/spock/tree/spock-2.0

**Comment by jacattau** [comment]
That is very nice to hear.  Thank you for the update.

**Comment by Carmen123456789** [comment]
Does the snapshot spock-core-2.0-groovy-2.5 support parallel for @Unroll now? Seems it can't work in my case. Or is there some  configuration to turn it on?

**Comment by Vampire** [comment]
Not yet, but Leonard is working on it at #1207.

**Comment by Carmen123456789** [comment]
> Not yet, but Leonard is working on it at #1207.

Thanks. So when could this be fixed even in snapshot? If it takes a long time, is there a workaround for this?

**Comment by leonard84** [comment]
@Carmen123456789 we are currently waiting on the JUnit 5.7-M2 release, which should be out this Friday. After that I hope to get it finished this month. There is currently no workaround.

**Comment by Carmen123456789** [comment]
> @Carmen123456789 we are currently waiting on the JUnit 5.7-M2 release, which should be out this Friday. After that I hope to get it finished this month. There is currently no workaround.

Thanks! Looking foward to it!

**Comment by roma0297** [comment]
Hi @leonard84, any updates on this?

**Comment by leonard84** [comment]
@roma0297 I'm waiting on a code review but you can check out the branch and give early feedback.

**Comment by leonard84** [comment]
#1207  has been merged to master, everyone interested (@roma0297, @Carmen123456789, ...) please try it out, hint read the docs http://spockframework.org/spock/docs/2.0-SNAPSHOT/parallel_execution.html
